### PR TITLE
Update CONTRIBUTING with merge commit policy and instructions.

### PR DESCRIPTION
**Pull request by anorth**
Requested by #2057.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -50,20 +71,18 @@ Check out the [Go-Filecoin code overview](CODEWALK.md) for a brief tour of the c
...
-### Code Reviews
+We review every change before merging to master, using GitHub pull requests.
+**Try to keep PRs small**, no more than 400 lines or 8 files. Code reviews are easier, faster, and more effective when the diff is small.
 
 `go-filecoin` requires 2 approvals for all PRs. We use the following process, which aims to merge code quickly and efficiently while avoiding both accidental and malicious introduction of bugs, unintended consequences, or vulnerabilities.
 
-The first review is done by a committer familiar with that area of the codebase. Once they deem it ready for merge, they will assign a maintainer for the second review. Once the PR has 2 Approvals and blocking comments have been addressed, the committer rebases and merges the PR.
+1. The first review is done by a committer familiar with that area of the codebase. Once they deem it ready for merge, they will assign a maintainer for the second review. 
+1. Once the PR has 2 Approvals and blocking comments have been addressed, the committer [rebases and merges](#landing-changes) the PR.
+    - If the author is also a committer, they merge their own PRs.
```
Comment (Kubuxu) [inline CONTRIBUTING.md]: `master` branch is currently protected, I think the only minimal number of people have merge access to it.

I think the group of commiters is much wider than a group of people with master merge permission.

Comment (mishmosh) [reply CONTRIBUTING.md]: > `master` branch is currently protected, I think the only minimal number of people have merge access to it.

@filecoin-project/go-filecoin-collaborators have write, but not merge access. This group is fairly broad, includes many people with a history of useful PRs.
@filecoin-project/go-filecoin-committers have merge access to master. This group is quite limited.

@anorth's wording is technically correct, but @Kubuxu is it confusing to you when you read it? Should we say "If the author has merge access as a committer or maintainer, they merge their own PRs"?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -74,14 +93,29 @@ We use the following conventions for code reviews:
   
 - By default, code review comments are advisory: the reviewee should consider them but doesn't _have_ to respond or address them. Comments that start with "BLOCKING" must be addressed and responded to. If a reviewer makes a blocking comment but does not block merging (by marking the review "Add Comments" or "Approve") then the reviewee can merge if the issue is addressed.
 
-In rare cases, a maintainter may request approval from all maintainers for a wide-reaching PR.
+In rare cases, a maintainer may request approval from all maintainers for a wide-reaching PR.
```
Comment (mishmosh) [inline CONTRIBUTING.md]: thx :)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -74,14 +93,29 @@ We use the following conventions for code reviews:
...
 
-#### Code Reviewer Responsibilities:
+#### Reviewer Responsibilities:
 
-**Avoid lengthy design discussions in PR reviews,** since major design questions should be addressed during the [Design Before Code](#design-before-code) step. If the conversation snowballs, prefer to merge and spin out an issue for discussion, and then follow up on the process failure that led to needing a big design discussion in a PR.
+**Avoid lengthy design discussions in PR reviews.** Major design questions should be addressed during the [Design Before Code](#design-before-code) step. If the conversation snowballs, prefer to merge and spin out an issue for discussion, and then follow up on the process failure that led to needing a big design discussion in a PR.
 
 It is considered helpful to add blocking comments to PRs that introduce protocol changes that do not appear in the [spec](#the-spec).
 
+### Landing Changes
```
Comment (mishmosh) [inline CONTRIBUTING.md]: nice section.

File: CONTRIBUTING.md (hunk 1 above)
Comment (anorth) [reply CONTRIBUTING.md]: Thanks @mishmosh your wording is more robust, I'll use that.

**Pull request merged**